runtime/compiler/runtime/RelocationRuntime.cpp

```diff
@@ -1326,7 +1326,12 @@ TR_SharedCacheRelocationRuntime::validateAOTHeader(TR_FrontEnd *fe, J9VMThread *
          {
          checkAOTHeaderFlags(hdrInCache, featureFlags);
          }
-      else if (hdrInCache->gcPolicyFlag != javaVM()->memoryManagerFunctions->j9gc_modron_getWriteBarrierType(javaVM()) )
+      else if (!((hdrInCache->gcPolicyFlag == javaVM()->memoryManagerFunctions->j9gc_modron_getWriteBarrierType(javaVM())) ||
+                 ((hdrInCache->gcPolicyFlag == gc_modron_wrtbar_cardmark_and_oldcheck) &&
```

Comment:
looks good, although a comment would be beneficial why it's compatible.
something like this or shorter: it's safe to run AOT code with inline cardmark barrier even if runtime will not need it since the barrier is only executed for active Concurrent Mark cycles (which won't ever happen with Concurrent Mark disabled in runtime)

Reply:
addressed as in the new push